### DISCUSSION TITLE
spock with TestNG and gradle

### DISCUSSION THREAD

**Discussion by hemanthsridhar** [spockframework/spock#1603] [Q&A]
Hi, I am trying to integrate spock with TestNG with gradle as the build management tool

class BlaTest extends Specification {
@org.testng.annotations.Test
def "xyz"(){
given:
        int x = 10

 when:
        int y = 10

 then:
        x+y == 10

}

}

build.gradle
test {
    useTestNG(){
        useDefaultListeners = true
    }
}

the test runs fine when invoked and assertion fails. However, the testname in reports are coming as "$spock_feature_"
How can we fix this? How can we get the correct method name.

Thank you.

**Comment by leonard84** [comment]
Spock 2.0 is a JUnit Platform based `TestEngine` and it doesn't work with `TestNG` you need to run it with `useJUnitPlatform()`.

**Comment by Vampire** [comment]
I'm closing this as it is a repost from Gitter message https://matrix.to/#/!LGXdVfHQijqsmVRhFs:gitter.im/$F5OQ_6SaPNVU_jOM4SMMuOj-kyT16JYInmzDllcjOig?web-instance[element.io]=app.gitter.im and discussion evolved more there already
